request.js

```diff
@@ -273,6 +275,9 @@ function emitOutboundCallsSent() {
 TChannelRequest.prototype.resend = function resend() {
     var self = this;
 
+    var isRetry = self.resendSanity < self.limit;
+    self.channel.incrementRequest(isRetry);
```

Comment:
This `channel` is the sub channel so this does mean we have on set of retry limits per outbound thing.

However just for sanity, let's move the logic into a new class and do `self.channel.x.increment()`

Then we can only allocate it conditionally like ( https://github.com/uber/tchannel-node/blob/acc7e9b126e6eba4d30ec9edaa8cd74a2f70cf70/channel.js#L187-L191 ). We can have it be `null` on the top channel.